src/diff/patch.js

```diff
@@ -82,6 +82,12 @@ export function patch(
 		// Once we have successfully rendered the new VNode, copy it's ID over
 		internal._vnodeId = newVNode._vnodeId;
 	} catch (e) {
+		if (e.then) {
+			// If a promise was thrown, reorderChildren in case this component is
+			// being hidden or moved
+			reorderChildren(internal, startDom, parentDom);
+		}
+
```

Comment:
This is the code that ensures a suspending component's DOM children are re-parented to the new disconnected DOM parent when mounting the fallback